src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -43,63 +46,207 @@ namespace intel_cpu {
 namespace node {
 namespace {
 
-/* This class implementation is a temporal WA
-   TODO: revise the implementation to remove the node reference*/
+struct SnippetKey {
+    Snippet::SnippetAttrs attrs;
+
+    size_t hash() const;
+    bool operator==(const SnippetKey& rhs) const;
+};
+
+size_t SnippetKey::hash() const {
+    using namespace dnnl::impl;
+    using namespace dnnl::impl::primitive_hashing;
+
+    size_t seed = 0;
+    for (const auto& blockedDim : attrs.inMemBlockedDims)
+        seed = get_vector_hash(seed, blockedDim);
+    for (const auto& order : attrs.inMemOrders)
+        seed = get_vector_hash(seed, order);
+    for (const auto& prec : attrs.inMemPrecs)
+        seed = hash_combine(seed, prec.getPrecVal());
+
+    for (const auto& blockedDim : attrs.outMemBlockedDims)
+        seed = get_vector_hash(seed, blockedDim);
+    for (const auto& order : attrs.outMemOrders)
+        seed = get_vector_hash(seed, order);
+    for (const auto& prec : attrs.outMemPrecs)
+        seed = hash_combine(seed, prec.getPrecVal());
+
+    seed = hash_combine(seed, attrs.bodyHash);
+
+    return seed;
+}
+
+bool SnippetKey::operator==(const SnippetKey& rhs) const {
+    if (attrs.bodyHash != rhs.attrs.bodyHash)
+        return false;
+    if (attrs.inMemBlockedDims.size() != rhs.attrs.inMemBlockedDims.size() ||
+        attrs.inMemOrders.size() != rhs.attrs.inMemOrders.size() ||
+        attrs.inMemPrecs.size() != rhs.attrs.inMemPrecs.size())
+        return false;
+    if (attrs.outMemBlockedDims.size() != rhs.attrs.outMemBlockedDims.size() ||
+        attrs.outMemOrders.size() != rhs.attrs.outMemOrders.size() ||
+        attrs.outMemPrecs.size() != rhs.attrs.outMemPrecs.size())
+        return false;
+
+    bool retVal = true;
+    for (size_t i = 0; i < attrs.inMemBlockedDims.size(); i++)
+        retVal = retVal && attrs.inMemBlockedDims[i] == rhs.attrs.inMemBlockedDims[i];
```

Comment:
Also probably we can improve the method if we order checks. I mean that `inMemPrecs` and `outMemPrecs` should be the end of method since precisions aren't changed after each infer request (or very rarely anyway?)? for example. @chenhu-wang what do you think?